library/src/main/java/com/bumptech/glide/RequestBuilder.java

```diff
@@ -359,15 +359,11 @@ protected RequestOptions getMutableOptions() {
     Request request = buildRequest(target);
 
     Request previous = target.getRequest();
-    // When request was failed or cancelled, be sure to use the updated model as it can contains
-    // unexposed data that could help the request to succeed on restart.
-    // See https://github.com/bumptech/glide/issues/2270
-    if (request.isEquivalentTo(previous)
-      && (Preconditions.checkNotNull(previous).isComplete()
-         || Preconditions.checkNotNull(previous).isRunning())) {
+    if (request.isEquivalentTo(previous)) {
       request.recycle();
       // If the request is completed, beginning again will ensure the result is re-delivered,
-      // triggering RequestListeners and Targets. If the request is already
+      // triggering RequestListeners and Targets. If the request is failed, beginning again will
```

Comment:
If the request has failed, calling into again will restart the request...

Reply:
Actually I restored your previous comment I think you were talking about the begin call after the if.